### PR TITLE
Capture empty tags

### PR DESCRIPTION
**Pull request by JuanFML**
# Description
Updated xml to capture empty tags in both python and JS files
- [x] Source branch in your fork has meaningful name (not `master`)

Fixes Issue:  #1854 

# Before Merge Checklist 
These items can be completed after PR is created.

(Check any items that are not applicable (NA) for this PR)

- [x] JavaScript implementation
- [x] Python implementation (NA if HTML beautifier)
- [x] Added Tests to data file(s)
- [ ] Added command-line option(s) (NA if
- [ ] README.md documents new feature/option(s)

### CODE REVIEW

File: test/data/javascript/tests.js [hunk 1]
```diff
@@ -2490,6 +2490,20 @@ exports.test_data = {
...
+        },
+        {
+          unchanged: [
+            'class Columns extends React.Component {',
+            '    render() {',
+            '        return (',
+            '            <a>',
+            '              <td>Hello</td>',
+            '              <td>World</td>',
+            '            </a>',
```
Comment (bitwiseman) [inline test/data/javascript/tests.js:2502]: This needs to be empty to actually test your change right? 
```suggestion
            '            <>',
            '              <td>Hello</td>',
            '              <td>World</td>',
            '            </>',
```

File: js/src/javascript/tokenizer.js [hunk 2]
```diff
@@ -126,7 +126,7 @@ var Tokenizer = function(input_string, options) {
     html_comment_end: pattern_reader.matching(/-->/),
     include: pattern_reader.starting_with(/#include/).until_after(acorn.lineBreak),
     shebang: pattern_reader.starting_with(/#!/).until_after(acorn.lineBreak),
-    xml: pattern_reader.matching(/[\s\S]*?<(\/?)([-a-zA-Z:0-9_.]+|{[\s\S]+?}|!\[CDATA\[[\s\S]*?\]\])(\s+{[\s\S]+?}|\s+[-a-zA-Z:0-9_.]+|\s+[-a-zA-Z:0-9_.]+\s*=\s*('[^']*'|"[^"]*"|{[\s\S]+?}))*\s*(\/?)\s*>/),
+    xml: pattern_reader.matching(/[\s\S]*?<(\/?)([-a-zA-Z:0-9_.]+|{[\s\S]+?}|!\[CDATA\[[\s\S]*?\]\]|)(\s+{[\s\S]+?}|\s+[-a-zA-Z:0-9_.]+|\s+[-a-zA-Z:0-9_.]+\s*=\s*('[^']*'|"[^"]*"|{[\s\S]+?}))*\s*(\/?)\s*>/),
```
Comment (bitwiseman) [inline js/src/javascript/tokenizer.js:129]: I think it is clearer to say this group is optional, but that might cause the number groups to change, right? 
```
    xml: pattern_reader.matching(/[\s\S]*?<(\/?)([-a-zA-Z:0-9_.]+|{[\s\S]+?}|!\[CDATA\[[\s\S]*?\]\])?(\s+{[\s\S]+?}|\s+[-a-zA-Z:0-9_.]+|\s+[-a-zA-Z:0-9_.]+\s*=\s*('[^']*'|"[^"]*"|{[\s\S]+?}))*\s*(\/?)\s*>/),
```

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Very nice change.

**Comment by JuanFML** [comment]
While fixing this issue I saw something that could be another issue:
In https://beautifier.io/ whenever you try to beautify a code with an indent of 2 spaces to an indent with 4 spaces it is unable to beautify it correctly:
**Input: (with predefined options only with Support e4x/jsx syntax selected)**
```
class Columns extends React.Component {
  render() {
    return (
      <t>        
        <td>Hello</td>
        <td>World</td>
      </t>
    );
  }
}
```
**Output**:
```
class Columns extends React.Component {
    render() {
        return (
            <t>        
        <td>Hello</td>
        <td>World</td>
      </t>
        );
    }
}
```
Although if I changed to indent to 2 spaces per indent, the output would be unchanged from the first input which is the expected output. This is because the input is tabbed with 2 spaces. Maybe the error is because of the single tags, although I haven't looked into it yet

**Comment by bitwiseman** [comment]
@JuanFML 
Yes, you're seeing #665.

File: test/data/javascript/tests.js (hunk 1 above)
Comment (JuanFML) [reply test/data/javascript/tests.js:2502]: Oh! I didn't see I hadn't changed it back, I will change this now

**Review by bitwiseman — APPROVED** [review]

File: js/src/javascript/tokenizer.js (hunk 2 above)
Comment (JuanFML) [reply js/src/javascript/tokenizer.js:129]: Oh I see what you mean, but yes it causes other problems

**Comment by JuanFML** [comment]
> @JuanFML
> Yes, you're seeing #665.

Oh yes that it is what I encounter, I see now it was already reported

**Comment by bitwiseman** [comment]
@Zetrick 
Not sure have your tried this on https://beautifier.io ? Looks good to me there.

**Pull request merged**